drivers/at86rf2xx/at86rf2xx_netdev.c

```diff
@@ -359,10 +359,17 @@ netopt_state_t _get_state(at86rf2xx_t *dev)
 
 static int _get(gnrc_netdev_t *device, netopt_t opt, void *val, size_t max_len)
 {
+    at86rf2xx_t *dev = (at86rf2xx_t *) device;
+
     if (device == NULL) {
         return -ENODEV;
     }
-    at86rf2xx_t *dev = (at86rf2xx_t *) device;
+
+    /* only getting state possible when device is sleeping */
+    if( (at86rf2xx_get_status(dev) == AT86RF2XX_STATE_SLEEP) &&
+        (opt != NETOPT_STATE) ) {
+        return -ECANCELED;
+    }
```

Comment:
I am not sure about this. Shouldn't we maybe wake the device up, to get/set options, and put it to sleep then again? I could imagine the MAC protocol awakes the device only when it expects receive/send actions and in this case it could be an advantage to be able to change options when there is nothing to do instead.